File: src/main/java/xyz/phanta/cbtweaker/buffer/ingredient/IngredientProviderType.java
package xyz.phanta.cbtweaker.buffer.ingredient;

import com.google.gson.JsonObject;
import net.minecraft.util.ResourceLocation;

public interface IngredientProviderType<A, JA> {

    ResourceLocation getId();

    IngredientProvider<A, JA> loadProvider(JsonObject config);

}

File: src/main/kotlin/st/evening/mc/cbtweaker/buffer/ingredient/IngredientProviderType.kt
package st.evening.mc.cbtweaker.buffer.ingredient

import st.evening.mc.cbtweaker.util.Identifiable
import st.evening.mc.prelude.api.data.tjson.JsonPath
import st.evening.mc.prelude.api.data.tjson.TJson

interface IngredientProviderType<A, JA> : Identifiable {
    context(_: JsonPath)
    fun loadProvider(dto: TJson.Object): IngredientProvider<A, JA>
}
